### ISSUE TITLE
Cannot read property '0' of undefined - ts source

### ISSUE THREAD

**Issue by krk** [quicktype/quicktype#799]
Large typescript source gives the error: `Cannot read property '0' of undefined` Unfortunately there is no line number. I can try to reduce the test case if I can have some clues :)

Input can be found at https://gist.github.com/krk/2c9f8aa8ecef1b4e05dd4c8b60d279f2

**Comment by schani** [comment]
Thank you!  I can't give a clue.  What I'd do to find the bug is run quicktype in the debugger.  Do you want me to write something up on how to do that?

**Comment by schani** [comment]
That bug is fixed now, but another bug prevents that file from working: https://github.com/YousefED/typescript-json-schema/issues/198

**Comment by schani** [comment]
I fixed the stack overflow, too, in PR #809 .  It's STILL not working, because of this type:

```typescript
interface IDBArrayKey extends Array<number | string | Date | IDBArrayKey> {
}
```

On the one hand, this is another bug in typescript-json-schema (https://github.com/YousefED/typescript-json-schema/issues/200), but even if it generated the correct Schema here, quicktype couldn't handle it, because of #564.

Defining this type in a way that's not a recursive array will produce a result.  If you have time to look at how correct the resulting types are, I'd be very interested.  Please file bugs for anything that's not as it should be.

This is as far as I'm willing to take this one at this point.  The recursive array issue isn't a big priority right now I think.
